### ISSUE TITLE
Parse of following example fail on Android artifact but not JVM

### ISSUE THREAD

**Issue by NitroG42** [pdvrieze/xmlutil#19]
Parsing the following line :

`<reclamation code="1"><![CDATA[30/06/2012 - M75 - Normal - NON LIVRE]]></reclamation>`

Using the following class:

```
@Serializable
@XmlSerialName(value = "reclamation", namespace = "", prefix = "")
data class Reclamation(@XmlValue(true) val content: String, val code: String)
```

works when using the jvm artifact (on a jvm machine), but on android it fails with the following error:
```

E/TestRunner: java.lang.AssertionError: Unexpected event in stream
        at nl.adaptivity.xmlutil.serialization.XmlDecoderBase$TagDecoder.decodeElementIndex(XMLDecoder.kt:705)
        at com.nitrog42.xmlserializationandroidbug.Reclamation$$serializer.deserialize(Unknown Source:19)
        at com.nitrog42.xmlserializationandroidbug.Reclamation$$serializer.deserialize(Reclamation.kt:9)
        at nl.adaptivity.xmlutil.serialization.XmlDecoderBase$XmlDecoder.decodeSerializableValue(XMLDecoder.kt:140)
        at nl.adaptivity.xmlutil.serialization.XML.parse(XML.kt:333)
        at nl.adaptivity.xmlutil.serialization.XML.parse(XML.kt:351)
        at com.nitrog42.xmlserializationandroidbug.TestReclamationAndroid.testParse(TestReclamationAndroid.kt:24)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:104)
        at org.junit.runners.Suite.runChild(Suite.java:128)
        at org.junit.runners.Suite.runChild(Suite.java:27)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
        at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
        at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
        at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:392)
        at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
```

This repository https://github.com/NitroG42/XmlSerializationAndroidBug reproduces the issue, you just need to run the androidTest or the test to get it.

It works if the CDATA part is replaced with a normal String.

**Comment by pdvrieze** [comment]
I've been able to replicate it in the main project (I've added a test). I'm not sure why it worked on JVM and not on Android as the bug is the serialization plugin not actually handling cdata content. Probably the jvm parser just makes it into text. In any case, it was a quick thing to handle cdata the same as regular text. It should be fixed by dcdb686c02096cc5417785295e8b273f7f6b26a2.

**Comment by pdvrieze** [comment]
New version 0.13.0.1 just released
